Initiating PR for new Azure AD Recommendations API (#19841)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/recommendation-list-impactedresources.md

---
title: "List impactedResources"

> description: "Get the impactedResource resources from the impactedResources navigation property."

Reviewer: ```suggestion
description: "Get the impacted resources for a recommendation."
```

author: "hafowler"
ms.localizationpriority: medium
ms.prod: "identity-and-access"
doc_type: apiPageType
---

# List impactedResources
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Get the impactedResource resources from the impactedResources navigation property.

Reviewer: ```suggestion
Get the [impactedResource](../resources/impactedresource.md) objects for a [recommendation](recommendation.md).
```

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|DirectoryRecommendations.Read.All, DirectoryRecommendations.ReadWrite.All|
|Delegated (personal Microsoft account)|Not supported.|
|Application|DirectoryRecommendations.Read.All, DirectoryRecommendations.ReadWrite.All|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
GET /directory/recommendations/{recommendationId}/impactedResources
```

## Optional query parameters

> This method supports some of the OData query parameters to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).

Reviewer: 👀

Reviewer: @hafowler - bump

          "additionalDetails": [
            {
              "key": "ExpiringCredentialsCount",
              "value": "1"
            }
          ]
        },
        {
          "id": "fd1482c2-3aab-4cad-8182-fc6adecf40ef",
          "subjectId": "ccec02c6-e69b-47d8-b6a0-c6cf9d491a4f",
          "recommendationId": "0cb31920-84b9-471f-a6fb-468c1a847088_Microsoft.Identity.IAM.Insights.ApplicationCredentialExpiry",
          "resourceType": "app",
          "addedDateTime": "2022-03-19T09:40:39.0413974Z",
          "postponeUntilDateTime": null,
          "lastModifiedDateTime": "2022-04-15T04:24:09.7862983Z",
          "lastModifiedBy": "0ffbbb05-658e-4f32-ae3e-7e717d7d9c91",
          "displayName": "Contoso",
          "owner": null,
          "rank": 1,
          "portalUrl": "https://portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/Credentials/appId/ccec02c6-e69b-47d8-b6a0-c6cf9d491a4f",
          "apiUrl": null,
          "status": "completedBySystem",
          "additionalDetails": [
            {
              "key": "ExpiringCredentialsCount",
              "value": "1"
            }
          ]
        }
    ]
}
```

> 

Reviewer: Add Example 2 using $select (at a minimum), $expand, and $filter

Reviewer: @hafowler - bump
